### ISSUE TITLE
prevChar isn't being defaulted when selecting an option when clicking

### ISSUE THREAD

**Issue by mjost5v** [fnagel/jquery-ui#97]
This causes problems when you switch between clicking and tabbing. In particular, it looks like the typeAhead doesn't work when clicking an option after typing ahead and then typing again. The suggested solution is to default the prevChar when opening the selectmenu.

I think I was able to resolve this issue by adding self._prevChar = ['', 0]; to the end of the open: function(event).

**Comment by fnagel** [comment]
Thanks for your contribution. Did you test this in several browsers? 

Could anybody do more tests here?

**Comment by mjost5v** [comment]
I tested this in Firefox 4, since we were doing UAT in that browser. I 
didn't get around to testing other browsers.

**Comment by fnagel** [comment]
Im not able to reproduce the issue. Tested in FF4 and latest stable Chrome.

What are the exact steps to reproduce the issue? Tabbing to a selectmenu, typeahead, click on the select so its open, typeahead again. Last typeahead should not work, right?

**Comment by mjost5v** [comment]
This issue wasn't easy to reproduce when the client first described it. 
It took a bit to figure out what the issue. I was using a dropdown 
consisting of US states and territories. The first option was blank as 
shown after the steps to recreate.

Here are the steps that I used:
1. Tab into a dropdown so that it's highlighted
2. Press a letter for typeahead to work
    I used 'V' so typeahead first went to Vermont
3. Keep on pressing that letter until you get to the end of the section 
   for that letter and typeahead doesn't go any further
    I kept on pressing 'V' to cycle through Virgin Islands and then 
   Virginia
4. Left click on the drop down to open it and select the blank option
5. The dropdown options will close but the dropdown itself will still be 
   highlighted.
6. Press the same character again for typeahead ('V' in my case). 
   Typeahead will not go anywhere.

<select id="state" >
<option></option>
<option value="AL">Alabama</option>
<option value="AK">Alaska</option>
<option value="AS">American Samoa</option>
<option value="AZ">Arizona</option>
<option value="AR">Arkansas</option>
<option value="CA">California</option>
<option value="CO">Colorado</option>
<option value="CT">Connecticut</option>
<option value="DE">Delaware</option>
<option value="DC">District of Columbia</option>
<option value="FM">Federated States of Micronesia</option>
<option value="FL">Florida</option>
<option value="GA">Georgia</option>
<option value="GU">Guam</option>
<option value="HI">Hawaii</option>
<option value="ID">Idaho</option>
<option value="IL">Illinois</option>
<option value="IN">Indiana</option>
<option value="IA">Iowa</option>
<option value="KS">Kansas</option>
<option value="KY">Kentucky</option>
<option value="LA">Louisiana</option>
<option value="ME">Maine</option>
<option value="MH">Marshall Islands</option>
<option value="MD">Maryland</option>
<option value="MA">Massachusetts</option>
<option value="MI">Michigan</option>
<option value="MN">Minnesota</option>
<option value="MS">Mississippi</option>
<option value="MO">Missouri</option>
<option value="MT">Montana</option>
<option value="NE">Nebraska</option>
<option value="NV">Nevada</option>
<option value="NH">New Hampshire</option>
<option value="NJ">New Jersey</option>
<option value="NM">New Mexico</option>
<option value="NY">New York</option>
<option value="NC">North Carolina</option>
<option value="ND">North Dakota</option>
<option value="MP">Northern Mariana Islands</option>
<option value="OH">Ohio</option>
<option value="OK">Oklahoma</option>
<option value="OR">Oregon</option>
<option value="PW">Palau</option>
<option value="PA">Pennsylvania</option>
<option value="PR">Puerto Rico</option>
<option value="RI">Rhode Island</option>
<option value="SC">South Carolina</option>
<option value="SD">South Dakota</option>
<option value="TN">Tennessee</option>
<option value="TX">Texas</option>
<option value="UT">Utah</option>
<option value="VT">Vermont</option>
<option value="VI">Virgin Islands</option>
<option value="VA">Virginia</option>
<option value="WA">Washington</option>
<option value="WV">West Virginia</option>
<option value="WI">Wisconsin</option>
<option value="WY">Wyoming</option>
</select>

**Comment by fnagel** [comment]
Sorry about the long waiting time, but I had an surgery and was not able to work on the selectmenu widget the last few weeks. I will take a look asap, please be patient!

**Comment by fnagel** [comment]
There has been some changes in the typeAhead algorithm so would you please test if this error still occurs when using this version: https://github.com/fnagel/jquery-ui/zipball/selectmenu_v1.1.0 or latest version

**Comment by mjost5v** [comment]
I retested with the latest code from GitHub and the select menu 
consisting of US states. The typeAhead algorithm doesn't go beyond the 
first option that starts with the letter anymore. For example, tabbing 
into the menu and then pressing 'N' once will cause it to go to 
Nebraska. Pressing 'N' again will keep it at Nebraska instead of going 
on to Nevada and other states that start with an 'N.'

After left clicking the menu and selecting the empty option, pressing 
'N' again will cause the menu to go back to Nebraska.

I've tested with other letters and gotten the same results. I can type 
at most 3 letters in quick succession before the prevChar variable (I'm 
guessing here. I didn't actually debug this time around) resets.

The issue is no longer valid due to the typeAhead algorithm changing.

Michael

**Comment by fnagel** [comment]
Thanks for your response!

Ive tested the typeAhead in a few browsers without any problems. Perhaps I should test with real word examples. Can you post your demo within a fiddle (see issue #61)? 

Did you test this in FF4 again? Please take a look here: https://github.com/fnagel/jquery-ui/pull/57
What do you mean with the last sentence?

**Comment by mjost5v** [comment]
I tested with Firefox 4 and Chrome. This is the selectmenu that I used:

<select id="state" >
<option></option>
<option value="AL">Alabama</option>
<option value="AK">Alaska</option>
<option value="AS">American Samoa</option>
<option value="AZ">Arizona</option>
<option value="AR">Arkansas</option>
<option value="CA">California</option>
<option value="CO">Colorado</option>
<option value="CT">Connecticut</option>
<option value="DE">Delaware</option>
<option value="DC">District of Columbia</option>
<option value="FM">Federated States of Micronesia</option>
<option value="FL">Florida</option>
<option value="GA">Georgia</option>
<option value="GU">Guam</option>
<option value="HI">Hawaii</option>
<option value="ID">Idaho</option>
<option value="IL">Illinois</option>
<option value="IN">Indiana</option>
<option value="IA">Iowa</option>
<option value="KS">Kansas</option>
<option value="KY">Kentucky</option>
<option value="LA">Louisiana</option>
<option value="ME">Maine</option>
<option value="MH">Marshall Islands</option>
<option value="MD">Maryland</option>
<option value="MA">Massachusetts</option>
<option value="MI">Michigan</option>
<option value="MN">Minnesota</option>
<option value="MS">Mississippi</option>
<option value="MO">Missouri</option>
<option value="MT">Montana</option>
<option value="NE">Nebraska</option>
<option value="NV">Nevada</option>
<option value="NH">New Hampshire</option>
<option value="NJ">New Jersey</option>
<option value="NM">New Mexico</option>
<option value="NY">New York</option>
<option value="NC">North Carolina</option>
<option value="ND">North Dakota</option>
<option value="MP">Northern Mariana Islands</option>
<option value="OH">Ohio</option>
<option value="OK">Oklahoma</option>
<option value="OR">Oregon</option>
<option value="PW">Palau</option>
<option value="PA">Pennsylvania</option>
<option value="PR">Puerto Rico</option>
<option value="RI">Rhode Island</option>
<option value="SC">South Carolina</option>
<option value="SD">South Dakota</option>
<option value="TN">Tennessee</option>
<option value="TX">Texas</option>
<option value="UT">Utah</option>
<option value="VT">Vermont</option>
<option value="VI">Virgin Islands</option>
<option value="VA">Virginia</option>
<option value="WA">Washington</option>
<option value="WV">West Virginia</option>
<option value="WI">Wisconsin</option>
<option value="WY">Wyoming</option>
</select>

Basically, what I meant with the last sentence in my previous email is
that I don't think this is an issue anymore. Although I didn't do any
debugging, it appears that prevChar is being reset in under a second.
When I filed the issue, prevChar wasn't being reset at all. Thus, you
could go to the end of the N's (for example), click an option starting
with a different character and have typeAhead still stuck at the end of
the N's.

Michael

**Comment by fnagel** [comment]
Thanks for your feedback and the nice demo code :-)
Any contribution is really appreciated and perhaps you will find some time to suggest some improvements. bergerb has some pending imporvenements too (see pull request mentioned above).

Closed as resolved for now.
